spring-cql/src/test/java/org/springframework/cassandra/config/PoolingOptionsFactoryBeanUnitTests.java

```diff
@@ -0,0 +1,229 @@
+	public void afterPropertiesSetInitializesRemotePoolingOptions() throws Exception {
+		PoolingOptionsFactoryBean poolingOptionsFactoryBean = new PoolingOptionsFactoryBean() {
+			@Override PoolingOptions newPoolingOptions() {
+				return mockPoolingOptions;
+			}
+		};
+
+		when(mockPoolingOptions.getNewConnectionThreshold(eq(HostDistance.REMOTE))).thenReturn(10);
+
+		poolingOptionsFactoryBean.setHeartbeatIntervalSeconds(30);
+		poolingOptionsFactoryBean.setIdleTimeoutSeconds(120);
+		poolingOptionsFactoryBean.setInitializationExecutor(mockExecutor);
+		poolingOptionsFactoryBean.setRemoteCoreConnections(5);
+		poolingOptionsFactoryBean.setRemoteMaxConnections(50);
+		poolingOptionsFactoryBean.setRemoteMaxSimultaneousRequests(20);
+		poolingOptionsFactoryBean.setRemoteMinSimultaneousRequests(5);
+		poolingOptionsFactoryBean.setPoolTimeoutMilliseconds(120);
+
+		assertThat(poolingOptionsFactoryBean.getObject(), is(nullValue(PoolingOptions.class)));
+
+		poolingOptionsFactoryBean.afterPropertiesSet();
+
+		assertThat(poolingOptionsFactoryBean.getObject(), is(sameInstance(mockPoolingOptions)));
+		assertThat(poolingOptionsFactoryBean.getObjectType(), is(equalTo((Class) mockPoolingOptions.getClass())));
+
+		InOrder inOrder = inOrder(mockPoolingOptions);
+
+		inOrder.verify(mockPoolingOptions, times(1)).setHeartbeatIntervalSeconds(eq(30));
+		inOrder.verify(mockPoolingOptions, times(1)).setIdleTimeoutSeconds(eq(120));
+		inOrder.verify(mockPoolingOptions, times(1)).setInitializationExecutor(eq(mockExecutor));
+		inOrder.verify(mockPoolingOptions, times(1)).setPoolTimeoutMillis(eq(120));
+		inOrder.verify(mockPoolingOptions, times(1)).setMaxConnectionsPerHost(eq(HostDistance.REMOTE), eq(50));
+		inOrder.verify(mockPoolingOptions, times(1)).setCoreConnectionsPerHost(eq(HostDistance.REMOTE), eq(5));
+		inOrder.verify(mockPoolingOptions, times(1)).setMaxRequestsPerConnection(eq(HostDistance.REMOTE), eq(20));
+		verify(mockPoolingOptions, never()).setCoreConnectionsPerHost(eq(HostDistance.LOCAL), anyInt());
+		verify(mockPoolingOptions, never()).setMaxConnectionsPerHost(eq(HostDistance.LOCAL), anyInt());
+		verify(mockPoolingOptions, never()).setMaxRequestsPerConnection(eq(HostDistance.LOCAL), anyInt());
+		verify(mockPoolingOptions, never()).setNewConnectionThreshold(eq(HostDistance.LOCAL), anyInt());
+		verify(mockPoolingOptions, never()).setNewConnectionThreshold(eq(HostDistance.REMOTE), eq(5));
+	}
+
+	/**
+	 * This particular test case is technically an integration test since it uses an actual instance of
+	 * a DataStax Java driver class type... {@link PoolingOptions}!
+	 *
+	 * The max values should be set before setting core values. Otherwise the core values will be compared with the
+	 * default max values which is 8. Same for other min-max properties pairs. This test checks the same.
+	 *
+	 * @throws Exception Any unhandled scenarios will result in a test failure.
+	 * @see <a href="https://jira.spring.io/browse/DATACASS-176">DATACASS-176</a>
```

Comment:
Are there options in the `Javadoc` Maven plugin to, perhaps, set `baseUrl` and/or `path` that could serve a placeholder for properties used in source?  (yuck) Though, saying that seems equally flawed since branch source will potentially become out-of-date if the URL ever does change.

Personally, I am not a big fan of improper syntax in `Javadoc`, so SD's use of the `@see` tags (i.e. @see DATACASS-xyz`) is rather broken.  I'd rather see no tag then an improper one.

Perhaps we can refer to JIRA tickets in another way, such as...

```
/**
 * Refer to DATACASS-xyz.
 */
```